### PR TITLE
feat: cc and bcc in ticket conversation reply

### PR DESCRIPTION
**Pull request by fadilsid**
Resolves https://github.com/frappe/helpdesk/issues/945

1.User will have the ability to add cc and bcc in ticket conversation reply.
2. Cc and bcc values will be displayed in ticket conversation

Frappe Desk (29)
Frappe Desk (28)
Frappe Desk (30)

### CODE REVIEW

**Comment by harshit-30** [comment]
@fadilsid the font size of the message body has changed to 16px, any reason why?

@esafwan are you reviewing the PRs?

File: desk/src/components/desk/ticket/ConversationCard.vue [hunk 1]
```diff
@@ -19,8 +19,15 @@
 				>{{ $dayjs.longFormating($dayjs(time).fromNow()) }}</a
 			>
 		</div>
-		<div class="pl-[32px] pt-[6px]">
+		<div class="pl-[32px] pt-[0px]">
```
Comment (netchampfaris) [inline desk/src/components/desk/ticket/ConversationCard.vue]: why use px values? why not `pl-8` and `pt-0`?

File: desk/src/components/desk/ticket/ConversationCard.vue [hunk 2]
```diff
@@ -19,8 +19,15 @@
...
 			<div class="flex flex-col">
+				<div
+					v-if="cc || bcc"
+					class="flex flex-row gap-1 text-[12px] text-[#74808B] font-inter"
```
Comment (netchampfaris) [inline desk/src/components/desk/ticket/ConversationCard.vue]: why explicit hex color?

File: desk/src/pages/desk/Ticket.vue [hunk 3]
```diff
@@ -109,22 +109,64 @@
...
+											class="flex flex-row space-x-2 items-center bg-transparent"
+										>
+											<div class="text-gray-700">Cc</div>
+											<Input
+												class="py-[4px] bg-white w-11/12 focus:bg-white text-[12px] font-inter pl-[4px]"
+												@input="
+													(val) => {
+														this.cc = val
+													}
+												"
```
Comment (netchampfaris) [inline desk/src/pages/desk/Ticket.vue]:
```suggestion
												@input="cc = $event"
```

File: desk/src/pages/desk/Ticket.vue [hunk 4]
```diff
@@ -649,21 +739,33 @@ export default {
...
+				bcc: this.bcc,
+				attachments: this.attachments.map((x) => x.name),
+			})
+			this.content = ""
+			this.attachments = []
+		},
+		submitResolvedTicket() {
+			this.tempTextEditorData.content = this.content
+			this.tempTextEditorData.attachments = this.attachments
+			const content = `<div class='content-block'><div>${this.content}</div></div>`
```
Comment (netchampfaris) [inline desk/src/pages/desk/Ticket.vue]: why is this required?

File: desk/src/components/desk/global/CustomIcons.vue [hunk 5]
```diff
@@ -1505,6 +1505,23 @@
 				stroke-linejoin="round"
 			/>
 		</svg>
+		<svg
+			v-if="this.name == 'arrow-down'"
+			:class="this.class"
+			:width="this.width"
+			:height="this.height"
```
Comment (netchampfaris) [inline desk/src/components/desk/global/CustomIcons.vue]:
```suggestion
			v-if="name == 'arrow-down'"
			:class="class"
			:width="width"
			:height="height"
```

**Review by netchampfaris — COMMENTED** [review]
bad code formatting, hardcoded values in tailwind classes, if this goes on the code quality will reduce with every PR

File: desk/src/components/desk/ticket/ConversationCard.vue (hunk 1 above)
Comment (fadilsid) [reply desk/src/components/desk/ticket/ConversationCard.vue]: found px values in frappedesk codebase. Will change this.

**Comment by fadilsid** [comment]
> @fadilsid the font size of the message body has changed to 16px, any reason why? Screenshot 2023-01-20 at 16 35 33
> 
> before this, I pointed out the whitespace issue that was fixed today.
> 
> @esafwan are you reviewing the PRs?

I haven't changed the font-size. it was 16 px before. Just checked it with the main branch.
Screenshot from 2023-01-23 10-49-02

**Comment by harshit-30** [comment]
@fadilsid My old branch reflects this as 13px, check the screenshot below: 
Screenshot 2023-01-23 at 17 35 41

Even if its 16px now, let's make it to 13.

**Comment by harshit-30** [comment]
@ssiyad thinking of reopening this issue for us to keep this in check - so that you can keep reviewing this in your task list, should we?
